dmaws/rds.py

```diff
@@ -295,20 +287,27 @@ def clean_database(self):
                     WHERE (
                         SELECT COUNT(*) FROM supplier_frameworks WHERE supplier_id=suppliers.supplier_id
                     ) = 0
+                    AND supplier_id NOT IN (
+                        SELECT DISTINCT supplier_id FROM services
+                    )
                 ), d1 AS (
                     DELETE FROM contact_information WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 ), d2 AS (
                     DELETE FROM users WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 )
-             DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
+            DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
         """)
 
+        # Remove audit events because we don't use them
         self.log("Delete audit events")
         self.cursor.execute("DELETE FROM audit_events")
 
+        # Remove framework agreements because they contain personal data and we don't rely on them
         self.log("Delete framework agreements")
         self.cursor.execute("DELETE FROM framework_agreements")
 
+        # Overwrite declarations with the smallest possible valid entry
```

Comment:
I've kept the current declaration and run through a few scenarios to see what breaks. 

1. signing an agreement as a supplier
2. putting an agreement on hold as an admin
3. countersigning an agreement as an admin

Turns out that nothing breaks.  We get some empty fields (including an empty link(!) for the companies' house address), but that doesn't actually cause an error.

This is the result: 
![screen shot 2017-01-25 at 15 45 51](https://cloud.githubusercontent.com/assets/2454380/22298398/4cd5267e-e318-11e6-92dc-95375f8a220c.png)

Personally, I'd argue that keeping the declaration as small as possible is more important than a fake address and a bad link, so if we're happy with this then I'll just merge this PR as-is.